File: frontend/src/types/bgg.ts
export interface BGGSearchResult {
  bgg_id: number;
  name: string;
  year_published: number | null;
}

export interface BGGGameDetails {
  bgg_id: number;
  name: string;
  description: string;
  image_url: string;
  thumbnail_url: string;
  min_players: number | null;
  max_players: number | null;
  min_playtime: number | null;
  max_playtime: number | null;
  year_published: number | null;
  min_age: number | null;
  weight: number | null;
  categories: string[];
  mechanics: string[];
  designers: string[];
  publishers: string[];
}

File: backend/app/schemas/bgg.py
from pydantic import BaseModel


class BGGSearchResult(BaseModel):
    bgg_id: int
    name: str
    year_published: int | None = None


class BGGGameDetails(BaseModel):
    bgg_id: int
    name: str
    description: str
    image_url: str
    thumbnail_url: str
    min_players: int | None
    max_players: int | None
    min_playtime: int | None
    max_playtime: int | None
    year_published: int | None
    min_age: int | None
    weight: float | None
    categories: list[str]
    mechanics: list[str]
    designers: list[str]
    publishers: list[str]


class BGGImportRequest(BaseModel):
    bgg_id: int
